geniushubclient/zoneclasses/properties.py

```diff
@@ -0,0 +1,78 @@
+from typing import Dict
+from geniushubclient.const import (
+    ITYPE_TO_TYPE,
+    TYPE_TO_ITYPE,
+    ZONE_KIT,
+    ZONE_TYPE,
+)
+
+
+class Properties:
+    """The properties of the zone.
+
+    Intended to be the properties that are relatively static for
+    the lifecycle of the zone"""
+
+    def __init__(self):
+        self.__name: str = None
+        self.__zone_type: int = None
+        self.__subtype: int = None
+        self.__type_name: str = None
+        self.__has_room_sensor: bool = None
+
+    def update(self, json_data: Dict, api_version: int) -> None:
+        """Parse the json and use it to populate the properties data class."""
+
+        if api_version == 1:
+            self._update_from_v1_data(json_data)
+        elif api_version == 3:
+            self._update_from_v3_data(json_data)
+
+    def _update_from_v1_data(self, json_data: Dict) -> None:
+        self.__name = json_data["name"]
+        self.__type_name = json_data["type"]
+        self.__zone_type = TYPE_TO_ITYPE[self.type_name]
+        self.__has_room_sensor = "occupied" in json_data
+
+    def _update_from_v3_data(self, json_data: Dict) -> None:
+        self.__name = json_data["strName"]
+        self.__zone_type = iType = json_data["iType"]
+        self.__subtype = json_data["zoneSubType"]
+        self.__has_room_sensor = json_data["iFlagExpectedKit"] & ZONE_KIT.PIR
+        self.__type_name = ITYPE_TO_TYPE[iType]
+
+        if iType == ZONE_TYPE.TPI and json_data["zoneSubType"] == 0:
+            self.__zone_type = ZONE_TYPE.ControlOnOffPID
+            self.__type_name = ITYPE_TO_TYPE[ZONE_TYPE.ControlOnOffPID]
+
+    def get_v1_data(self) -> Dict:
```

Comment:
The get_v1_data method? If I make that private, then how is the zone going to populate the v1 json with the values in properties?

Reply:
Sorry - I think 'private' wasn't specific enough. It is a pseudo-slang term, and probably means something different in other languages.

Specifically, I mean, should we call it `_get_v1_data` - the leading underscore just sends a message: 
>The underscore prefix is meant as a hint to another programmer that a variable or method starting with a single underscore is intended for internal use. This convention is defined in PEP 8. 

Here, 'internal' means for the library's own use.

When I say it hasn't got legs, I mean it won't be persistent enough for other devs to rely upon it as a public method. They _can_ use it if they want, tho'.

c.f. Mangling.

Reply:
Yeah, in C# private means that nothing outside of the class can even see the property. What you are talking about is literally marked with internal in C#.

But, you're right, this method is not intended for use outside of the library. And, actually, neither is the update method in this class, that should only be called by the genius client, as it will have made the API call, so I'll update that to internal too. If we end up finding a use case for making it public, then that's an easy enough change to make later, and as that is an additive change, it's not breaking the public contract.